show/muxcable.py

```diff
@@ -0,0 +1,160 @@
+            platform_sfputil.read_porttab_mappings(port_config_file_path, 0)
+    except Exception as e:
+        log.log_error("Error reading port info (%s)" % str(e), True)
+        sys.exit(3)
+
+    return
+
+
+@muxcable.command()
+@click.argument('port', required=False, default=None)
+def status(port):
+    """Show muxcable summary information"""
+
+    state_db = {}
+    y_cable_tbl = {}
+    port_table_keys = {}
+
+    # Getting all front asic namespace and correspding config and state DB connector
+
+    namespaces = multi_asic.get_front_end_namespaces()
+    for namespace in namespaces:
+        asic_id = multi_asic.get_asic_index_from_namespace(namespace)
+        state_db[asic_id] = swsscommon.DBConnector("STATE_DB", REDIS_TIMEOUT_MSECS, True, namespace)
+        y_cable_tbl[asic_id] = swsscommon.Table(state_db[asic_id], "HW_MUX_CABLE_TABLE_NAME")
+        port_table_keys[asic_id] = y_cable_tbl[asic_id].getKeys()
+
+    if port is not None:
+        asic_index = platform_sfputil.get_asic_id_for_logical_port(port)
+        click.echo("asic_index {} \n".format(asic_index))
+        if asic_index is None:
+            click.echo("Got invalid asic index for port {}, cant retreive mux status".format(port))
+            return
+
+        y_cable_asic_table = y_cable_tbl.get(asic_index, None)
+        if y_cable_asic_table is not None:
+            y_cable_asic_table_keys = y_cable_asic_table.getKeys()
+            if port in y_cable_asic_table_keys:
+                port_status_dict = {}
+                port_status_dict["mux_cable"] = {}
+                (status, fvs) = y_cable_asic_table.get(str(port))
+                if status is not True:
+                    click.echo("could not retrieve port values for port".format(port))
+                fvp = dict(fvs)
+                status_value = fvp.get("status", None)
+                port_status_dict["mux_cable"][port] = {"status": status_value}
+
+                click.echo("muxcable Ports status : \n {}".format(json.dumps(port_status_dict, indent=4)))
+            else:
+                click.echo("this is not a valid port present on mux_cable".format(port))
+        else:
+            click.echo("there is not a valid asic table for this asic_index".format(asic_index))
+
+    else:
+
+        port_status_dict = {}
+        port_status_dict["mux_cable"] = {}
+        for namespace in namespaces:
+            asic_id = multi_asic.get_asic_index_from_namespace(namespace)
+            for key in port_table_keys[asic_id]:
+                (status, fvs) = y_cable_tbl[asic_id].get(key)
+                if status is not True:
+                    click.echo("could not retrieve port values for port".format(key))
+                fvp = dict(fvs)
+                status_value = fvp.get("status", None)
+                port_status_dict["mux_cable"][key] = {"status": status_value}
+
+        click.echo("muxcable Ports status : \n {}".format(json.dumps(port_status_dict, indent=4)))
```

Comment:
I think we should consider allowing a `--json` flag to output results in JSON format, otherwise output in a human-readable tabular format using `tabulate`.

Reply:
fixed